x-pack/plugin/security/src/test/java/org/elasticsearch/integration/ClusterPrivilegeTests.java

```diff
@@ -203,5 +205,12 @@ private void waitForSnapshotToFinish(String repo, String snapshot) throws Except
             SnapshotsStatusResponse response = client().admin().cluster().prepareSnapshotStatus(repo).setSnapshots(snapshot).get();
             assertThat(response.getSnapshots().get(0).getState(), is(SnapshotsInProgress.State.SUCCESS));
         });
+        // The status of the snapshot in the repository can become SUCCESS before it is fully finalized in the cluster state so wait for
+        // it to disappear from the cluster state as well
+        assertBusy(() -> {
+            SnapshotsInProgress response =
+                client().admin().cluster().state(new ClusterStateRequest()).get().getState().custom(SnapshotsInProgress.TYPE);
+            assertThat(response.entries(), Matchers.empty());
+        });
```

Comment:
@original-brownbear you've got my rubber stamp LGTM , but shouldn't we fix the `prepareSnapshotStatus(repo).setSnapshots(snapshot)` API to not return `SUCCESS` if there is still an `SnapshotsInProgress.TYPE` entry?